### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by cuishuang**

### CODE REVIEW

File: lib/libc/include/any-macos-any/dispatch/io.h [hunk 1]
```diff
@@ -549,7 +549,7 @@ dispatch_io_set_high_water(dispatch_io_t channel, size_t high_water);
  *
  * I/O handlers should be prepared to receive amounts of data significantly
  * larger than the low water mark in general. If an I/O handler requires
- * intermediate results of fixed size, set both the low and and the high water
+ * intermediate results of fixed size, set both the low andthe high water
```
Comment (rofrol) [inline lib/libc/include/any-macos-any/dispatch/io.h:552]: and the

File: lib/libc/include/any-macos-any/xpc/connection.h [hunk 2]
```diff
@@ -303,7 +303,7 @@ xpc_connection_set_target_queue(xpc_connection_t connection,
  * A connection may receive different events depending upon whether it is a
  * listener or not. Any connection may receive an error in its event handler.
  * But while normal connections may receive messages in addition to errors,
- * listener connections will receive connections and and not messages.
+ * listener connections will receive connections andnot messages.
```
Comment (rofrol) [inline lib/libc/include/any-macos-any/xpc/connection.h:306]: and not

File: lib/std/fmt/parse_float/convert_eisel_lemire.zig [hunk 3]
```diff
@@ -79,7 +79,7 @@ pub fn convertEiselLemire(comptime T: type, q: i64, w_: u64) ?BiasedFp(f64) {
     }
 
     // Need to handle rounding ties. Normally, we need to round up,
-    // but if we fall right in between and and we have an even basis, we
+    // but if we fall right in between andwe have an even basis, we
```
Comment (rofrol) [inline lib/std/fmt/parse_float/convert_eisel_lemire.zig:82]: and we

File: src/link/Wasm.zig [hunk 4]
```diff
@@ -926,7 +926,7 @@ fn resolveSymbolsInArchives(wasm: *Wasm) !void {
             };
 
             // Symbol is found in unparsed object file within current archive.
-            // Parse object and and resolve symbols again before we check remaining
+            // Parse object andresolve symbols again before we check remaining
```
Comment (rofrol) [inline src/link/Wasm.zig:929]: and resolve

**Comment by Vexu** [comment]
We don't need to fix typos in vendored code (`lib/libc`).

**Pull request closed without merging**